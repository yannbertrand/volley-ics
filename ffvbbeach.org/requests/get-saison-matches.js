import Papa from 'papaparse'
import getRequestBody from '../body.builder.js'

export async function getSaisonMatches(client, saison, ent, poule) {
  let response
  try {
    response = await client.request({
      path: '/ffvbapp/resu/vbspo_calendrier_export.php',
      method: 'POST',
      headers: {
        'content-type': 'application/x-www-form-urlencoded',
      },
      body: getRequestBody({
        typ_edition: 'E',
        type: 'RES',
        cal_saison: saison,
        cal_codent: ent,
        cal_codpoule: poule,
      }),
    })
  } catch (exception) {
    handleFfvbbRequestError(exception)
  }

  const csvContent = await response.body.text()
  const rawContent = Papa.parse(csvContent, {
    header: true,
    skipEmptyLines: true,
    transformHeader: (header) =>
      header.startsWith('Entit') ? 'Entite' : header,
  }).data

  const mapEquipeA = new Map()
  mapEquipeA.set('id', 'EQA_no')
  mapEquipeA.set('nom', 'EQA_nom')

  const mapEquipeB = new Map()
  mapEquipeB.set('id', 'EQB_no')
  mapEquipeB.set('nom', 'EQB_nom')

  const mapArbitres = new Set()
  mapArbitres.add('Arb1')
  mapArbitres.add('Arb2')

  const mapMatch = new Map()
  mapMatch.set('entite', 'Entite')
  mapMatch.set('journee', 'Jo')
  mapMatch.set('matchId', 'Match')
  mapMatch.set('date', 'Date')
  mapMatch.set('heure', 'Heure')
  mapMatch.set('equipeA', mapEquipeA)
  mapMatch.set('equipeB', mapEquipeB)
  mapMatch.set('set', 'Set')
  mapMatch.set('score', 'Score')
  mapMatch.set('total', 'Total')
  mapMatch.set('salle', 'Salle')
  mapMatch.set('arbitres', mapArbitres)

  return mapData(rawContent, mapMatch)
}

function handleFfvbbRequestError(exception) {
  switch (exception.errno) {
    case -3008:
      throw new Error(
        'www.ffvbbeach.org was not found, are you connected to the internet?\nOr do you need to work with mocked data using `USE_NETWORK=false`?'
      )
    default:
      console.error(exception)
      throw new Error(
        'An unkown error occured while requesting www.ffvbbeach.org'
      )
  }
}

function mapData(data, mapValues) {
  return data.map((item) => {
    const mappedData = {}

    for (const [toKey, fromKey] of mapValues) {
      if (fromKey instanceof Map) {
        const subData = {}
        for (const [subKey, realFromKey] of fromKey) {
          subData[subKey] = item[realFromKey]
        }
        mappedData[toKey] = subData
        continue
      }

      if (fromKey instanceof Set) {
        const subData = []
        for (const realFromKey of fromKey) {
          subData.push(item[realFromKey])
        }
        mappedData[toKey] = subData
        continue
      }

      if (toKey === 'salle') {
        mappedData['salle'] = getSalleAdress(item[fromKey].trim())
      } else {
        mappedData[toKey] = item[fromKey].trim()
      }
    }

    return mappedData
  })
}

function getSalleAdress(salle) {
  return (
    {
      'SALLE LONGCHAMP': 'Salle Longchamp, 42 Rue Chanteclerc, 44300 Nantes',
      'SALLE DU CHEVALIER':
        'Salle du Chevalier, 44450 Saint-Julien-de-Concelles',
      'SALLE GARDIN': 'Salle Jean-Pierre Gardin, 44860 Pont St. Martin',
      'SALLE RAPHAEL LEBEL':
        'Gymnase Raphaël Lebel, 16 Bd Auguste Peneau, 44300 Nantes',
      'LA SENSIVE':
        'Gymnase De La Sensive, Rue de Bordeaux, 44800 Saint-Herblain',
      'SALLE DE LA GAGNERIE':
        'Salle de la gagnerie, 17 Rue de la Gagnerie, 44340 Bouguenais',
      'SALLE GASTON TURPIN':
        'Gymnase Gaston Turpin, 33 Rue Gaston Turpin, 44000 Nantes',
      'GYMNASE L. ET M. LIBERGE':
        'Gymnase Liberge, 11 Rue de la Galarnière, 44400 Rezé',
      'SALLE DE LA PAPINIERE':
        'Salle de la Papinière, 18 Rue de la Papinière, 44240 Sucé-sur-Erdre',
    }[salle] || salle
  )
}
