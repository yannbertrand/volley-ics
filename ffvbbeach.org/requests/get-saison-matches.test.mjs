import { describe, beforeEach, it, expect } from 'vitest'
import getMockedFfvbbClient from '../mock.mjs'
import { getSaisonMatches } from './get-saison-matches.mjs'

describe('getSaisonMatches', () => {
  let ca1
  beforeEach(async () => {
    client = await getMockedFfvbbClient()
    ca1 = await getSaisonMatches(client, '2021/2022', 'PTPL44', 'CA1')
  })

  it('should fetch ffvbbeach.org data', () => {
    expect(ca1).toBeDefined()
    expect(ca1).toHaveLength(45)
    expect(ca1).toMatchSnapshot()
  })
})
