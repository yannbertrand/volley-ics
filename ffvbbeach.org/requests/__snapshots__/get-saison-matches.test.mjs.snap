// Vitest Snapshot v1

exports[`getSaisonMatches > should fetch ffvbbeach.org data 1`] = `
[
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-01",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "equipeB": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "heure": "20:00",
    "journee": "01",
    "matchId": "CA1A001",
    "salle": "GYMNASE GASTON TURPIN",
    "score": "25-22,25-22,25-18",
    "set": "3/0",
    "total": "75-62",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-09-28",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "equipeB": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "heure": "20:50",
    "journee": "01",
    "matchId": "CA1A002",
    "salle": "GYMNASE BOTTIERE CHESNAIE",
    "score": "12-25,19-25,21-25",
    "set": "0/3",
    "total": "52-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-09-27",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "heure": "20:30",
    "journee": "01",
    "matchId": "CA1A003",
    "salle": "SALLE GARDIN",
    "score": "25-18,25-16,25-20",
    "set": "3/0",
    "total": "75-54",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-09-27",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "equipeB": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "heure": "20:30",
    "journee": "01",
    "matchId": "CA1A004",
    "salle": "SALLE LONGCHAMP",
    "score": "25-20,15-25,16-25,24-26",
    "set": "1/3",
    "total": "80-96",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-09-29",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "heure": "20:15",
    "journee": "01",
    "matchId": "CA1A005",
    "salle": "GYMNASE M. ET M. LIBERGE",
    "score": "25-20,22-25,25-21,17-25,13-15",
    "set": "2/3",
    "total": "102-106",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-05",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "heure": "21:00",
    "journee": "02",
    "matchId": "CA1A006",
    "salle": "SALLE DE LA GAGNERIE",
    "score": "25-23,16-25,25-18,25-20",
    "set": "3/1",
    "total": "91-86",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-06",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "equipeB": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "heure": "20:00",
    "journee": "02",
    "matchId": "CA1A007",
    "salle": "SALLE DE LA PAPINIERE",
    "score": "20-25,25-17,25-21,17-25,14-16",
    "set": "2/3",
    "total": "101-104",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-04",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "heure": "20:30",
    "journee": "02",
    "matchId": "CA1A008",
    "salle": "SALLE LONGCHAMP",
    "score": "25-13,25-18,25-11",
    "set": "3/0",
    "total": "75-42",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-06",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "heure": "20:00",
    "journee": "02",
    "matchId": "CA1A009",
    "salle": "LA SENSIVE",
    "score": "22-25,19-25,23-25",
    "set": "0/3",
    "total": "64-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-08",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "equipeB": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "heure": "20:00",
    "journee": "02",
    "matchId": "CA1A010",
    "salle": "GYMNASE GASTON TURPIN",
    "score": "25-13,25-7,25-14",
    "set": "3/0",
    "total": "75-34",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-12",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "equipeB": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "heure": "20:50",
    "journee": "03",
    "matchId": "CA1A011",
    "salle": "GYMNASE BOTTIERE CHESNAIE",
    "score": "20-25,20-25,17-25",
    "set": "0/3",
    "total": "57-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-11",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "heure": "20:30",
    "journee": "03",
    "matchId": "CA1A012",
    "salle": "SALLE GARDIN",
    "score": "18-25,22-25,20-25",
    "set": "0/3",
    "total": "60-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-11",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "equipeB": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "heure": "20:30",
    "journee": "03",
    "matchId": "CA1A013",
    "salle": "SALLE LONGCHAMP",
    "score": "25-21,10-25,25-15,16-25,11-15",
    "set": "2/3",
    "total": "87-101",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-13",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "heure": "20:15",
    "journee": "03",
    "matchId": "CA1A014",
    "salle": "GYMNASE M. ET M. LIBERGE",
    "score": "19-25,18-25,25-22,25-18,8-15",
    "set": "2/3",
    "total": "95-105",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-13",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "equipeB": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "heure": "20:00",
    "journee": "03",
    "matchId": "CA1A015",
    "salle": "SALLE RAPHAEL LEBEL",
    "score": "15-25,28-26,25-20,25-20",
    "set": "3/1",
    "total": "93-91",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-19",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "equipeB": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "heure": "21:00",
    "journee": "04",
    "matchId": "CA1A016",
    "salle": "SALLE DE LA GAGNERIE",
    "score": "25-15,24-26,20-25,25-18,15-13",
    "set": "3/2",
    "total": "109-97",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-18",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "heure": "20:30",
    "journee": "04",
    "matchId": "CA1A017",
    "salle": "SALLE LONGCHAMP",
    "score": "25-15,17-25,20-25,25-16,11-15",
    "set": "2/3",
    "total": "98-96",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-20",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "equipeB": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "heure": "20:00",
    "journee": "04",
    "matchId": "CA1A018",
    "salle": "LA SENSIVE",
    "score": "17-25,25-19,25-21,25-19",
    "set": "3/1",
    "total": "92-84",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-22",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "heure": "20:00",
    "journee": "04",
    "matchId": "CA1A019",
    "salle": "GYMNASE GASTON TURPIN",
    "score": "25-13,25-12,25-15",
    "set": "3/0",
    "total": "75-40",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-10-19",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "heure": "20:50",
    "journee": "04",
    "matchId": "CA1A020",
    "salle": "GYMNASE BOTTIERE CHESNAIE",
    "score": "18-25,14-25,15-25",
    "set": "0/3",
    "total": "47-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-15",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "equipeB": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "heure": "20:30",
    "journee": "05",
    "matchId": "CA1A021",
    "salle": "SALLE GARDIN",
    "score": "23-25,25-21,18-25,19-25",
    "set": "1/3",
    "total": "85-96",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-15",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "equipeB": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "heure": "20:30",
    "journee": "05",
    "matchId": "CA1A022",
    "salle": "SALLE LONGCHAMP",
    "score": "19-25,21-25,20-25",
    "set": "0/3",
    "total": "60-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-17",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "heure": "20:15",
    "journee": "05",
    "matchId": "CA1A023",
    "salle": "GYMNASE M. ET M. LIBERGE",
    "score": "17-25,25-21,12-25,18-25",
    "set": "1/3",
    "total": "72-96",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-17",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "equipeB": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "heure": "20:00",
    "journee": "05",
    "matchId": "CA1A024",
    "salle": "SALLE RAPHAEL LEBEL",
    "score": "25-20,25-21,23-25,25-19",
    "set": "3/1",
    "total": "98-85",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-17",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "heure": "20:00",
    "journee": "05",
    "matchId": "CA1A025",
    "salle": "SALLE DE LA PAPINIERE",
    "score": "0-25,0-25,0-25",
    "set": "0/3",
    "total": "0-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-23",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "heure": "21:00",
    "journee": "06",
    "matchId": "CA1A026",
    "salle": "SALLE DE LA GAGNERIE",
    "score": "25-17,25-15,18-25,18-25,13-15",
    "set": "2/3",
    "total": "99-97",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-24",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "equipeB": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "heure": "20:00",
    "journee": "06",
    "matchId": "CA1A027",
    "salle": "LA SENSIVE",
    "score": "25-20,27-25,28-26",
    "set": "3/0",
    "total": "80-71",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-26",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "heure": "20:00",
    "journee": "06",
    "matchId": "CA1A028",
    "salle": "GYMNASE GASTON TURPIN",
    "score": "31-29,25-20,23-25,17-25,16-18",
    "set": "2/3",
    "total": "112-117",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-23",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "equipeB": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "heure": "20:50",
    "journee": "06",
    "matchId": "CA1A029",
    "salle": "GYMNASE BOTTIERE CHESNAIE",
    "score": "25-23,18-25,25-21,14-25,8-15",
    "set": "2/3",
    "total": "90-109",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-22",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "heure": "20:30",
    "journee": "06",
    "matchId": "CA1A030",
    "salle": "SALLE GARDIN",
    "score": "25-18,25-12,25-20",
    "set": "3/0",
    "total": "75-50",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-29",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "equipeB": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "heure": "20:30",
    "journee": "07",
    "matchId": "CA1A031",
    "salle": "SALLE LONGCHAMP",
    "score": "15-25,23-25,21-25",
    "set": "0/3",
    "total": "59-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-01",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "heure": "20:15",
    "journee": "07",
    "matchId": "CA1A032",
    "salle": "GYMNASE M. ET M. LIBERGE",
    "score": "17-25,19-25,21-25",
    "set": "0/3",
    "total": "57-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-01",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "equipeB": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "heure": "20:00",
    "journee": "07",
    "matchId": "CA1A033",
    "salle": "SALLE RAPHAEL LEBEL",
    "score": "25-20,25-20,25-20",
    "set": "3/0",
    "total": "75-60",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-01",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "heure": "20:00",
    "journee": "07",
    "matchId": "CA1A034",
    "salle": "SALLE DE LA PAPINIERE",
    "score": "16-25,11-25,13-25",
    "set": "0/3",
    "total": "40-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-11-29",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "equipeB": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "heure": "20:30",
    "journee": "07",
    "matchId": "CA1A035",
    "salle": "SALLE LONGCHAMP",
    "score": "22-25,25-16,25-19,25-15",
    "set": "3/1",
    "total": "97-75",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-07",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "equipeB": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "heure": "21:00",
    "journee": "08",
    "matchId": "CA1A036",
    "salle": "SALLE DE LA GAGNERIE",
    "score": "18-25,25-22,25-22,25-16",
    "set": "3/1",
    "total": "93-85",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-10",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "heure": "20:00",
    "journee": "08",
    "matchId": "CA1A037",
    "salle": "GYMNASE GASTON TURPIN",
    "score": "25-19,17-25,25-14,25-16",
    "set": "3/1",
    "total": "92-74",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-07",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "equipeB": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "heure": "20:50",
    "journee": "08",
    "matchId": "CA1A038",
    "salle": "GYMNASE BOTTIERE CHESNAIE",
    "score": "13-25,25-18,24-26,16-25",
    "set": "1/3",
    "total": "78-94",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-06",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "heure": "20:30",
    "journee": "08",
    "matchId": "CA1A039",
    "salle": "SALLE GARDIN",
    "score": "25-20,25-12,25-12",
    "set": "3/0",
    "total": "75-44",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-06",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "equipeB": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "heure": "20:30",
    "journee": "08",
    "matchId": "CA1A040",
    "salle": "SALLE LONGCHAMP",
    "score": "23-25,25-22,25-12,25-14",
    "set": "3/1",
    "total": "98-73",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-15",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442443",
      "nom": "ASBR 1",
    },
    "equipeB": {
      "id": "0448821",
      "nom": "US BOUGUENAIS 1",
    },
    "heure": "20:15",
    "journee": "09",
    "matchId": "CA1A041",
    "salle": "GYMNASE M. ET M. LIBERGE",
    "score": "25-23,20-25,25-18,18-25,10-15",
    "set": "2/3",
    "total": "98-106",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-15",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0440018",
      "nom": "CASSAV",
    },
    "equipeB": {
      "id": "0442446",
      "nom": "ASPTT 2",
    },
    "heure": "20:00",
    "journee": "09",
    "matchId": "CA1A042",
    "salle": "SALLE RAPHAEL LEBEL",
    "score": "25-20,25-20,21-25,25-20",
    "set": "3/1",
    "total": "96-85",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-15",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0446750",
      "nom": "JGE SUCE 1",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "US PONT ST MARTIN",
    },
    "heure": "20:00",
    "journee": "09",
    "matchId": "CA1A043",
    "salle": "SALLE DE LA PAPINIERE",
    "score": "29-27,14-25,25-22,11-25,9-15",
    "set": "2/3",
    "total": "88-114",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-13",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0442446",
      "nom": "ASPTT 1",
    },
    "equipeB": {
      "id": "0440024",
      "nom": "NPVB 1",
    },
    "heure": "20:30",
    "journee": "09",
    "matchId": "CA1A044",
    "salle": "SALLE LONGCHAMP",
    "score": "25-15,25-7,25-13",
    "set": "3/0",
    "total": "75-35",
  },
  {
    "arbitres": [
      "",
      "",
    ],
    "date": "2021-12-15",
    "entite": "PTPL44",
    "equipeA": {
      "id": "0448335",
      "nom": "ST HERBLAIN VB 2",
    },
    "equipeB": {
      "id": "0440018",
      "nom": "LEO VOLLEY DETENTE 1",
    },
    "heure": "20:00",
    "journee": "09",
    "matchId": "CA1A045",
    "salle": "LA SENSIVE",
    "score": "22-25,17-25,25-17,22-25",
    "set": "1/3",
    "total": "86-92",
  },
]
`;
